### PR TITLE
Adding credentials set support demisto rest api

### PR DESCRIPTION
**Pull request by YuvHayun**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
added credentials for the api-key key and relevant tests.

## Does it break backward compatibility?
   - [ ] Yes
   - [X] No

### CODE REVIEW

File: Packs/DemistoRESTAPI/Integrations/DemistoRESTAPI/DemistoRESTAPI.yml [hunk 1]
```diff
@@ -11,14 +11,20 @@ configuration:
   defaultvalue: https://127.0.0.1
   type: 0
   required: true
-- display: Demisto Server API Key
-  name: apikey
-  type: 4
-  required: true
+- displaypassword: Demisto Server API Key
+  name: api_key
```
Comment (ShahafBenYakir) [inline Packs/DemistoRESTAPI/Integrations/DemistoRESTAPI/DemistoRESTAPI.yml:15]: More meaningful
```suggestion
  name: creds_apikey
```

File: Packs/DemistoRESTAPI/Integrations/DemistoRESTAPI/DemistoRESTAPI.js [hunk 2]
```diff
@@ -15,13 +15,16 @@ sendMultipart = function (uri, entryID, body) {
         // do nothing, use the body as is in the request.
         logDebug('could not parse body as a JSON object, passing as is. body: ' + JSON.stringify(body));
     }
-
+    var key = [params.apikey? params.apikey : (params.api_key? params.api_key.password : '')];
```
Comment (ShahafBenYakir) [inline Packs/DemistoRESTAPI/Integrations/DemistoRESTAPI/DemistoRESTAPI.js:18]: Why do we do it this way and not like this?
```suggestion
    var key = [params.apikey? params.apikey :  params.api_key.password ];
```
Existing users will have `params.apikey`. New users will have `params.api_key`, such that doing `params.api_key.password` will return the API key or 'undefined', but will not error.

Comment (YuvHayun) [reply Packs/DemistoRESTAPI/Integrations/DemistoRESTAPI/DemistoRESTAPI.js:18]: I tried this, but if no api-key is provided then the user will get the following error:
'TypeError: Cannot read property 'password' of undefined at script'

Comment (ShahafBenYakir) [reply Packs/DemistoRESTAPI/Integrations/DemistoRESTAPI/DemistoRESTAPI.js:18]: How could this be? Doesn't `api_key` always exist? even if  `password`  was not filled?

Comment (YuvHayun) [reply Packs/DemistoRESTAPI/Integrations/DemistoRESTAPI/DemistoRESTAPI.js:18]: It exist, but if password isn't filled then api_key will hold null value and therefore trying to access .password will raise an error.

Comment (ShahafBenYakir) [reply Packs/DemistoRESTAPI/Integrations/DemistoRESTAPI/DemistoRESTAPI.js:18]: Roger

**Review by ShahafBenYakir — COMMENTED** [review]
I think you are missing this https://github.com/demisto/content/pull/17506#discussion_r802565411

File: Packs/DemistoRESTAPI/Integrations/DemistoRESTAPI/DemistoRESTAPI.yml (hunk 1 above)
Comment (YuvHayun) [reply Packs/DemistoRESTAPI/Integrations/DemistoRESTAPI/DemistoRESTAPI.yml]: Done.

**Review by ShahafBenYakir — APPROVED** [review]

**Comment by ShahafBenYakir** [comment]
@YuvHayun I already approved this 9 days ago. Fix the build errors and merge.

**Pull request merged**